dbms/src/Functions/FunctionsMath.h

```diff
@@ -716,38 +666,135 @@ struct PowName
     static constexpr auto name = "pow";
 };
 
+#pragma push_macro("UNARY_FUNCTION_IMPL")
+#pragma push_macro("UNARY_FUNCTION_IMPL_NORMAL")
+#define UNARY_FUNCTION_IMPL(X, FUNC)                                                \
+    struct X##FunctionImpl : public MathFunctionVectorizedTag                       \
+    {                                                                               \
+        static constexpr auto name = X##Name::name;                                 \
+        static constexpr auto rows_per_iteration = 1;                               \
+        template <typename T>                                                       \
+        static void transform(const T * src, Float64 * dst, size_t size)            \
+        {                                                                           \
+            ::DB::MathVectorization::UnaryMath::FUNC##Transform<T>(src, dst, size); \
+        }                                                                           \
+        template <typename T>                                                       \
+        static void execute(const T * src, Float64 * dst)                           \
+        {                                                                           \
+            dst[0] = static_cast<Float64>(::FUNC(static_cast<Float64>(src[0])));    \
+        }                                                                           \
+    }
+
+#define UNARY_FUNCTION_IMPL_NORMAL(X, FUNC) \
+    using X##FunctionImpl = UnaryFunctionPlain<X##Name, FUNC>
+
+#if defined(__linux__) && defined(__aarch64__)
+UNARY_FUNCTION_IMPL(Log, log);
+UNARY_FUNCTION_IMPL_NORMAL(Log2, log2);
```

Comment:
add comments to explain why we only make parts of the functions be vectorized? This paragraph is copied from the issue https://github.com/pingcap/tiflash/issues/5352
```
- vectorized the following functions on `x86-64` using `libmvec`(`libmvec`'s `exp` performs slower, so we do not use it):
   - `cos`
   - `sin`
   - `tan`
   - `acos`
   - `asin`
   - `atan`
   - `log`
   - `log2`
   - `cbrt`
- vectorize the following functions on `aarch64` using `optimized-routines`:
   - `cos`
   - `sin`
   - `log`
   - `exp`
```

Reply:
added comments